scripts/screenshot.js

```diff
@@ -6,21 +7,47 @@ const puppeteer = require('puppeteer');
 /
 */
 async function run() {
-    const executable = process.argv[4] || '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome'
-    const browser = await puppeteer.launch({ executablePath: executable, headless: true });
-    try {
-        const page = await browser.newPage();
-        await page.setViewport({ width: 1200, height: 800 });
-        await page.goto(process.argv[2], {
-          timeout: 30000, 
-          waitUntil: 'networkidle2'
+  const executable = process.argv[4] || '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome';
+  const browser = await puppeteer.launch(
+    {
+      executablePath: executable,
+      headless: true,
+      args: [
+        '--disable-web-security',
+        '--disable-features=IsolateOrigins',
+        '--disable-site-isolation-trials'
+      ]
+    }
+  );
+  const uri = process.argv[2]
+  const page = await browser.newPage();
+  try {
+    await page.setViewport({ width: 1200, height: 800 });
+    await page.goto(uri, {
+      timeout: 30000,
+      waitUntil: 'networkidle2'
+    });
+    await page.screenshot({ path: process.argv[3], format: 'jpeg' });
+  } catch (err) {
+    // Puppeteer cannot screenshot PDFs in headless mode by itself, so get some help from EJS and PDF.js
+    // HT: https://stackoverflow.com/a/70437748
+    if (err.message.match('net::ERR_ABORTED') && uri.match(/pdf/i)) {
+      try {
+        const html = await ejs.renderFile('./scripts/template.ejs', { data: { uri } });
+        await page.setContent(html, {
+          waitUntil: 'networkidle0',
+          timeout: 30000
         });
         await page.screenshot({ path: process.argv[3], format: 'jpeg' });
-    } catch (err) {
-        console.error(err.message)
-    } finally {
-        await browser.close()
+      } catch (err) {
+        console.error(err.message);
```

Comment:
I think the program should exit so that the workflow knows screenshot failed here? `process.exit(1)` ?

Reply:
I didn't realize this was in a busted state. Do you think we could take this up in a separate PR, or is it really as simple as adding those two `process.exitCode` lines? I'm flexible.

Reply:
My bad. It doesn't look like [the code](https://github.com/sul-dlss/was_robot_suite/blob/main/lib/dor/was_seed/thumbnail_generator_service.rb#L31) calling screeshot.js does anything with the exit code. It's using the presence of the screenshot jpg file, which seems good enough for now.

Reply:
ok!